math/src/unsigned_integer/element.rs

```diff
@@ -440,23 +440,29 @@ impl<const NUM_LIMBS: usize> UnsignedInteger<NUM_LIMBS> {
 
     /// Creates an `UnsignedInteger` from a hexstring
     /// # Panics
-    /// Panics if value is not a hexstring. Shouldn't start with `0x`
+    /// Panics if value is not a hexstring. It can contain `0x` or not.
     pub const fn from_hex_unchecked(value: &str) -> Self {
         let mut result = [0u64; NUM_LIMBS];
         let mut limb = 0;
         let mut limb_index = NUM_LIMBS - 1;
         let mut shift = 0;
-        let value = value.as_bytes();
-        let mut i: usize = value.len();
-        while i > 0 {
-            i -= 1;
-            limb |= match value[i] {
-                c @ b'0'..=b'9' => (c as u64 - '0' as u64) << shift,
-                c @ b'a'..=b'f' => (c as u64 - 'a' as u64 + 10) << shift,
-                c @ b'A'..=b'F' => (c as u64 - 'A' as u64 + 10) << shift,
-                _ => {
-                    panic!("Malformed hex expression.")
-                }
+
+        let value_bytes = value.as_bytes();
+
+        // Remove "0x" if it's at the beginning of the string
+        let mut i = 0;
+        if value_bytes.len() > 2 && value_bytes[0] == b'0' && value_bytes[1] == b'x' {
+            i = 2;
+        }
```

Comment:
I think it's better to use `starts_with` in this case:
https://doc.rust-lang.org/std/primitive.slice.html#method.starts_with

Reply:
@pablodeymo As `from_hex_unchecked` is a const function it will not call a non-const function. so is there any other way we can implement this

Reply:
Maybe:
```rust
if value_bytes.len() > 2 && value_bytes[..2] == b'0x' {
```

Reply:
I don't think this is going to work either compiler is giving this result.
```rust
    |   
454 |         if value_bytes.len() > 2 && value_bytes[..2] == b"0x" {
    |                                                      ^^ no implementation for `[u8] == &[u8; 2]`
```
alternatively if i try something like this 
```rust
454 |         if value_bytes.len() > 2 && value_bytes[..2] == [b'0', b'x'] 
```
it won't work as stated previously that `from_hex_unchecked` can't call a non const operator.

Reply:
no it still isn't working.